hazelcast-client/src/test/java/com/hazelcast/client/test/TestHazelcastFactory.java

```diff
@@ -68,39 +64,30 @@ public HazelcastInstance newHazelcastClient(ClientConfig config) {
             config = new XmlClientConfigBuilder().build();
         }
 
-        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
-        HazelcastClientProxy proxy;
+        Thread currentThread = Thread.currentThread();
+        ClassLoader tccl = currentThread.getContextClassLoader();
         try {
             if (tccl == ClassLoader.getSystemClassLoader()) {
-                Thread.currentThread().setContextClassLoader(HazelcastClient.class.getClassLoader());
+                currentThread.setContextClassLoader(HazelcastClient.class.getClassLoader());
             }
-            ClientConnectionManagerFactory clientConnectionManagerFactory =
-                    clientRegistry.createClientServiceFactory("127.0.0.1", CLIENT_PORTS);
-            AddressProvider testAddressProvider = createAddressProvider(config);
-            HazelcastClientInstanceImpl client =
-                    new HazelcastClientInstanceImpl(config, clientConnectionManagerFactory, testAddressProvider);
+            HazelcastClientInstanceImpl client = new HazelcastClientInstanceImpl(config,
+                    clientRegistry.createClientServiceFactory(), createAddressProvider(config));
             client.start();
             clients.add(client);
             OutOfMemoryErrorDispatcher.registerClient(client);
-            proxy = new HazelcastClientProxy(client);
+            return new HazelcastClientProxy(client);
         } finally {
-            Thread.currentThread().setContextClassLoader(tccl);
+            currentThread.setContextClassLoader(tccl);
         }
-        return proxy;
     }
 
     private AddressProvider createAddressProvider(ClientConfig config) {
         boolean discoveryEnabled = new HazelcastProperties(config.getProperties())
                 .getBoolean(ClientProperty.DISCOVERY_SPI_ENABLED);
-
         ClientAwsConfig awsConfig = config.getNetworkConfig().getAwsConfig();
-
         List<String> userConfiguredAddresses = config.getNetworkConfig().getAddresses();
 
-        boolean isAtLeastAProviderConfigured =
-                discoveryEnabled || (awsConfig != null ? awsConfig.isEnabled() : false) || !userConfiguredAddresses.isEmpty();
-
-        if (isAtLeastAProviderConfigured) {
+        if (discoveryEnabled || (awsConfig != null && awsConfig.isEnabled()) || !userConfiguredAddresses.isEmpty()) {
```

Comment:
TBH I don't really find this useful. Since the condition is quite long, the previous approach has nice advantage - you can tell right away from the variable's name, what it means. I would vote for the previous approach then.

Reply:
Okay, I'll revert the change.